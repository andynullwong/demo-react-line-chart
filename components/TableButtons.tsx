import { useFilterContext } from "@/context/filter.context";
import ChartRange from "@/types/chartRange.type";
import { Dispatch, SetStateAction } from "react";

const Button = ({
  filter,
  setFilter,
}: {
  filter: ChartRange;
  setFilter: Dispatch<SetStateAction<ChartRange>>;
}) => (
  <button
    type="button"
    className="relative inline-flex items-center rounded-l-md bg-white px-3 py-2 text-sm font-semibold text-gray-900 ring-1 ring-inset ring-gray-300 hover:bg-gray-50 focus:z-10"
    onClick={() => setFilter(filter)}
  >
    {filter}
  </button>
);

const TableButtons = () => {
  const { setFilter } = useFilterContext();

  return (
    <span className="isolate inline-flex rounded-md shadow-sm">
      {Object.keys(ChartRange).map((k) => (
        <Button
          key={k}
          filter={ChartRange[k as keyof typeof ChartRange]}
          setFilter={setFilter}
        />
      ))}
    </span>
  );
};

export default TableButtons;
