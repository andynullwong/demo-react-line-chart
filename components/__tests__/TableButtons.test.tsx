import React from "react";
import { render } from "@testing-library/react";
import TableButtons from "@/components/TableButtons";

test("TableButtons snapshot", () => {
  const { container } = render(<TableButtons />);
  expect(container).toMatchSnapshot();
});
