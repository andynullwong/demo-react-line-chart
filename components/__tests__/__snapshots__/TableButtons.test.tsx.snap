// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableButtons snapshot 1`] = `
<div>
  <span
    class="isolate flex justify-end rounded-md shadow-sm"
  >
    <button
      class="relative -ml-px inline-flex items-center px-3 py-2 text-sm font-semibold ring-1 ring-inset focus:z-10 text-gray-900 bg-gray-50 ring-gray-300"
      type="button"
    >
      ALL
    </button>
    <button
      class="relative -ml-px inline-flex items-center px-3 py-2 text-sm font-semibold ring-1 ring-inset focus:z-10 text-gray-900 bg-gray-50 ring-gray-300"
      type="button"
    >
      YTD
    </button>
    <button
      class="relative -ml-px inline-flex items-center px-3 py-2 text-sm font-semibold ring-1 ring-inset focus:z-10 text-gray-900 bg-gray-50 ring-gray-300"
      type="button"
    >
      12M
    </button>
    <button
      class="relative -ml-px inline-flex items-center px-3 py-2 text-sm font-semibold ring-1 ring-inset focus:z-10 text-gray-900 bg-gray-50 ring-gray-300"
      type="button"
    >
      3M
    </button>
    <button
      class="relative -ml-px inline-flex items-center px-3 py-2 text-sm font-semibold ring-1 ring-inset focus:z-10 text-gray-900 bg-gray-50 ring-gray-300"
      type="button"
    >
      1M
    </button>
  </span>
</div>
`;
